registry/endpoint.go

```diff
@@ -171,12 +168,12 @@ func (e *Endpoint) pingV1(factory *requestdecorator.RequestFactory) (RegistryInf
 		return RegistryInfo{Standalone: false}, nil
 	}
 
-	req, err := factory.NewRequest("GET", e.Path("_ping"), nil)
+	req, err := http.NewRequest("GET", e.Path("_ping"), nil)
 	if err != nil {
 		return RegistryInfo{Standalone: false}, err
 	}
 
-	resp, _, err := doRequest(req, nil, ConnectTimeout, e.IsSecure)
+	resp, err := HTTPClient(DefaultTransport(ConnectTimeout, e.IsSecure)).Do(req)
```

Comment:
Should this client have been passed in, rather than instantiated here? Or does it not require the same `RoundTripper`?